### DISCUSSION TITLE
drain cleaner with pdb

### DISCUSSION THREAD

**Discussion by Rockawear** [Opster/opensearch-k8s-operator#660] [Q&A]
Hi,

Is there a way to drain data pods before Karpenter evicts the pods? I'd like to know how this is handled if that feature exists in the  OpenSearch operator

**Comment by swoehrl-mw** [comment]
Hi @Rockawear. The operator currently has no special support for Karpenter aside from allowing you to define PodDisruptionBudgets for nodepools. The operator will only drain pods when it is performing updates/restarts itself.

**Reply by Rockawear** [reply]
Thanks for responding. Okay, setting PodDisruptionBudgets to maxUnavailable 0 does help with downtime, but not a solution. I'm wondering if admission webhook can help here. Do you know if this use case is in the roadmap?

**Reply by swoehrl-mw** [reply]
There is nothing planned or on the roadmap in regards to this. But if someone wants to contribute improvements in that direction we will likely accept them.
